lib/application_insights/rack/track_request.rb

```diff
@@ -54,7 +58,7 @@ def call(env)
           :url => request.url
         }
 
-        @client.track_request id, start_time, duration, status, success, options
+        @client.track_request request_id, start_time, duration, status, success, options
```

Comment:
Just to point out, if you want to do the correlation manually it will work. But for the UI components (AppMap, TraceViewer) in Azure portal, this won't work.
The reason it will first filter the telemetries by operationId, which is the first segement of the request id. Which means the operation context needs to be set properly.

Reply:
@mattrayner do you want to address this feedback? Setting operation_id will enable UI scenarios. I assume you want them to work

Reply:
@SergeyKanzhelev @yantang-msft I'll add this functionality tonight, I'm assuming it is as simple as passing operation_id as one of the options values?

Reply:
By operation_id I mean this context property: https://github.com/Microsoft/ApplicationInsights-Ruby/blob/develop/lib/application_insights/channel/contracts/operation.rb#L10

Reply:
Do you have an example of how I should pass this into the telemetry client?

When I look at this call, I can't see how I would pass this in?
https://github.com/mattrayner/ApplicationInsights-Ruby/blob/mattrayner/48_expose-request-id-within-requests/lib/application_insights/rack/track_request.rb#L61

Reply:
@mattrayner It's not that straightforward to pass it in currently. If you really want to do it, you need to call this https://github.com/Microsoft/ApplicationInsights-Ruby/blob/develop/lib/application_insights/telemetry_client.rb#L220 yourself, so you have the control over the RequestData and Context.
I'm fine with keeping it like this, anyways correlation is not going to work without track_dependency. We can focus on making correlation work automatically later.

Reply:
@yantang-msft I'm looking at the operation context stuff now, do I need to populate all of the attributes Foran operation, root, parent etc, or am I able to just provide an ID?

If I do need to provide them all, where is the best place to look for documentation on how this should look?

Reply:
Never mind, I found it here